src/core_codemods/with_threading_lock.py

```diff
@@ -39,8 +40,11 @@ def leave_With(self, original_node: cst.With, updated_node: cst.With):
         if len(original_node.items) == 1 and self.node_is_selected(
             original_node.items[0]
         ):
-            # TODO: how to avoid name conflicts here?
-            name = cst.Name(value="lock")
+            current_names = self.find_used_names_in_module()
```

Comment:
Does this reflect the current state of the module or just the original module? In other words, if we add a new variable with our codemod, will it show up in this list upon the next invocation?

Reply:
These are currently calculated based on the current module (obtained via the wrapper in `CodemodContext`) so any newly added variables should show up in the next in invocation. 
However, since this is a visitor, it may make several changes to the same module and thus the `current_names` may not contain the newly added names (recall that no changes are made to the module until after the visitor ends) in the next `leave_with` call.